### PR TITLE
Add a TypeScript example and tests

### PR DESCRIPTION
**Pull request by kwwxis**
Hello! Thanks for making this library, I've found it useful while developing locally.

It took me a good hour or two to figure out how to use it with TypeScript though, so I think adding an example could be helpful to others. I also added two tests for Vue components with TypeScript and ran the tests for 2.6, 2.7 and 3 to confirm they pass.

---

For the transform method, I was initially trying to do it like this:

```js
import tsLoader from 'esm-loader-typescript';
...
transform(source, ctx) {
  if (!vueTsRegex.test(ctx.url)) {
    return;
  }
  return tsLoader.transform(source, tx);
}
```

But to my confusion, that didn't work. So I looked in the `esm-loader-typescript` source code and found it only processes files ending with `.ts`, whereas the files that this loader needs to process would end with `.vue?vue&type=script&lang=ts`. So I had to settle for copying the code from `esm-loader-typescript`.

### CODE REVIEW

File: tsconfig.json
Comment (sebamarynissen) [inline tsconfig.json:1]: Move this to a separate TypeScript example. I would go with `test/typescript` as folder name.

**Comment by sebamarynissen** [comment]
Hi there, thanks for this! I'd be happy to merge this example, but I would prefer to put the tests in a separate folder called `test/typescript` or something. That way the root of the repository doesn't get polluted with `tsconfig.json`, which I'd like to avoid as the library doesn't really use TypeScript. Having a `tsconfig.json` in the root could be confusing for people.

**Comment by kwwxis** [comment]
Good point! I've made the change.

**Review by sebamarynissen — COMMENTED** [review]
Lgtm 👍

**Pull request merged**